### PR TITLE
Support for cancellable query

### PR DESCRIPTION
**Pull request by cheungpat**
connects #277 

Skygear Server is now built with golang 1.8.1.

### CODE REVIEW

File: pkg/server/skydb/pq/pq.go [hunk 1]
```diff
@@ -71,14 +72,14 @@ func isNetworkError(err error) bool {
 	return ok
 }
 
-type queryxRunner interface {
-	Exec(query string, args ...interface{}) (sql.Result, error)
-	Queryx(query string, args ...interface{}) (*sqlx.Rows, error)
-	QueryRowx(query string, args ...interface{}) *sqlx.Row
+type queryxContextRunner interface {
```
Comment (rickmak) [inline pkg/server/skydb/pq/pq.go]: What is the use of this interface...?

Comment (cheungpat) [inline pkg/server/skydb/pq/pq.go]: Interesting question. This interface ceased to have any use in skydb.

**Comment by cheungpat** [comment]
> Will we provide default timeout context?

When router handle a request, it creates a context with a timeout of 60s and this context is passed the the database connection struct in the conn_db preprocessor. So database query will be cancelled when the request is timed out.

File: pkg/server/skydb/pq/pq.go (hunk 1 above)
Comment (rickmak) [inline pkg/server/skydb/pq/pq.go]: Let's remove it?

Comment (cheungpat) [inline pkg/server/skydb/pq/pq.go]: Interesting. I thought I pushed something.

**Pull request merged**